File: Iniciante/3301.py
h, z, l = map(int, input().split())
if h > z and h > l and z < l:
    meio = 'luisinho'
elif h > z and h > l and z > l:
    meio = 'zezinho'
elif z > h and z > l and h > l:
    meio = 'huguinho'
elif z > h and z > l and l > h:
    meio = 'luisinho'
elif l > h and l > z and h > z:
    meio = 'huguinho'
elif l > h and l > z and z > h:
    meio = 'zezinho'
print(meio)

File: 3301.cpp
#include <iostream>
#include <string>

using namespace std;

int main(){
    int h, z, l;
    string meio;
    cin >> h >> z >> l; 
    if (h > z && h > l && z < l){
        meio = "luisinho";
    } else if (h > z && h > l && z > l){
        meio = "zezinho";
    }else if (z > h && z > l && h > l){
        meio = "huguinho";
    } else if (z > h && z > l && l > h){
        meio = "luisinho";
    } else if (l > h && l > z && h > z){
        meio = "huguinho";
    } else if (l > h && l > z && z > h){
        meio = "zezinho";
    }
    cout << meio << endl;
    return 0;
}